File: app/src/main/java/org/skyfaced/mvp/di/LocalModule.java
package org.skyfaced.mvp.di;

import org.skyfaced.mvp.service.instant.InstantService;
import org.skyfaced.mvp.service.instant.InstantServiceImpl;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import dagger.hilt.InstallIn;
import dagger.hilt.components.SingletonComponent;

@Module
@InstallIn(SingletonComponent.class)
public final class LocalModule {
    @Provides
    @Singleton
    InstantService provideInstanceService() {
        return new InstantServiceImpl();
    }
}

File: app/src/main/java/org/skyfaced/mvp/di/LocalModule.kt
package org.skyfaced.mvp.di

import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.components.SingletonComponent
import org.skyfaced.mvp.service.instant.InstantService
import org.skyfaced.mvp.service.instant.InstantServiceImpl
import javax.inject.Singleton

@Module
@InstallIn(SingletonComponent::class)
object LocalModule {
    @Provides
    @Singleton
    fun provideInstanceService(): InstantService = InstantServiceImpl()
}